src/particle_filter/src/particleFilter.cpp

```diff
@@ -142,6 +142,17 @@ void particleFilter::addObservation(double obs[2][3], vector<vec4x3> &mesh, dist
   cout << "Total time: " << total_time << endl;
   cout << "Average time: " << total_time / 20.0 << endl << endl;
 
+  ofstream myfile;
+  myfile.open("/home/shiyuan/Documents/ros_marsarm/diff.csv", ios::out|ios::app);

```

Comment:
Hardcoded path
I'm not going to merge this in, as it would create problem running on my machine